### PR TITLE
Improved mobile UX

### PR DESCRIPTION
**Pull request by sunify**

### CODE REVIEW

File: src/routes/transaction.tsx [hunk 1]
```diff
@@ -90,23 +94,28 @@ class Transaction extends React.Component<TransactionRouteProps, any> {
     return (
       <Card title="Transaction" className="explorer-section">
         <h3>Hash</h3>
-        {this.tx.hash}
+        <HexString />
```
Comment (troggy) [inline src/routes/transaction.tsx]: is this needed?

File: src/style.css [hunk 2]
```diff
@@ -110,4 +114,76 @@ h2:first-child {
...
+  width: 100% !important;
+}
+
+.wallet-input .ant-form-item-control-wrapper {
+  width: 100%;
+}
+
+@media screen and (min-width: 500px) {
+  .wallet-input {
+    width: 450px;
```
Comment (troggy) [inline src/style.css]: I would remove this:
a) it is not necessary
b) doesn't works nice with viewport width 500–576

**Comment by troggy** [comment]
@sunify pushed couple of suggestions here. Please feel free to adjust

File: src/components/hexString.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const { Fragment } = React;
+
+interface HexStringProps {}
+
+const HexString: React.SFC<HexStringProps> = ({ children }) => {
+  return (
+    <Fragment>
+      <span className="leap-hex-full">{children}</span>
+      <span className="leap-hex-short" title={children as string}>
```
Comment (troggy) [inline src/components/hexString.tsx]: maybe use `react-responsive` here? We already have it bundled

Comment (troggy) [reply src/components/hexString.tsx]: will make things cleaner. However, I'm not aware of performance downside

I would then use three media queries here:
min-width 760px — show full
min-width 500px — show shortened if hex length > 60 (for txHashes)
< 500px — show shortened

Comment (sunify) [reply src/components/hexString.tsx]: Nice idea

**Pull request merged**